consensus/src/common.rs

```diff
@@ -176,6 +176,20 @@ impl Consensus for ProtocolStateChain {
                 // short-range fork, select longer chain
                 self.select_longer_chain(candidate)
             } else {
+                // check against sub window density sizes > 11
+                let tip_state = self
+                    .consensus_state()
+                    .ok_or(ConsensusError::ConsensusStateNotFound)?;
+                let candidate_state = candidate
+                    .consensus_state()
+                    .ok_or(ConsensusError::ConsensusStateNotFound)?;
+
+                if tip_state.sub_window_densities.len() != 11 {
+                    return Ok(candidate);
+                } else if candidate_state.sub_window_densities.len() != 11 {
+                    return Ok(self);
+                }
+
                 let tip_density = self.relative_min_window_density(candidate)?;
                 let candidate_density = candidate.relative_min_window_density(self)?;
```

Comment:
Since we are already checking the window densities len here, the error won't be propagated to the caller from `relative_min_window_density`
code suggestion:
> let tip_density = match self.relative_min_window_density(candidate) {
Ok(d) => Ok(candidate),
Err(err) => // Maybe log this error
}
let candidate_density = match candidate.relative_min_window_density(self) {
Ok(d) => Ok(candidate),
Err(err) => // Maybe log this error
}

Reply:
Yeah, the idea is to not even consider computing the relative density if the window length is not of expected length.

Reply:
then can we remove len check from `relative_min_window_density`?

```
if tip_state.sub_window_densities.len() != 11 || chain_b.sub_window_densities.len() != 11 {
            return Err(ConsensusError::InvalidSubWindowDensityLen);
        }
```

Reply:
Yeah, this will be removed